pkg/kubectl/cmd/templates/templates.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import "strings"
+
+func MainHelpTemplate() string {
+	return decorate(mainHelpTemplate, false)
+}
+
+func MainUsageTemplate() string {
+	return decorate(mainUsageTemplate, true) + "\n"
+}
+
+func OptionsHelpTemplate() string {
+	return decorate(optionsHelpTemplate, false)
+}
+
+func OptionsUsageTemplate() string {
+	return decorate(optionsUsageTemplate, false)
+}
+
+func decorate(template string, trim bool) string {
+	if trim && len(strings.Trim(template, " ")) > 0 {
+		template = strings.Trim(template, "\n")
+	}
+	return template
+}
+
+const (
+	vars = `{{$isRootCmd := isRootCmd .}}` +
+		`{{$rootCmd := rootCmd .}}` +
+		`{{$visibleFlags := visibleFlags (flagsNotIntersected .LocalFlags .PersistentFlags)}}` +
+		`{{$explicitlyExposedFlags := exposed .}}` +
+		`{{$optionsCmdFor := optionsCmdFor .}}` +
+		`{{$usageLine := usageLine .}}`
+
+	mainHelpTemplate = `{{with or .Long .Short }}{{. | trim}}{{end}}{{if or .Runnable .HasSubCommands}}{{.UsageString}}{{end}}`
+
+	mainUsageTemplate = vars + `{{if gt .Aliases 0}}
+Aliases:
+  {{.NameAndAliases}}{{end}}{{if .HasExample}}
+
+Examples:
+{{ .Example | trimRight}}
+{{end}}{{ if .HasAvailableSubCommands}}{{range cmdGroups . .Commands}}
+{{.Message}}{{range .Commands}}{{if .Runnable}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}
+{{end}}{{end}}{{ if or $visibleFlags.HasFlags $explicitlyExposedFlags.HasFlags}}
+Options:
+{{ if $visibleFlags.HasFlags}}{{flagsUsages $visibleFlags}}{{end}}{{ if $explicitlyExposedFlags.HasFlags}}{{flagsUsages $explicitlyExposedFlags}}{{end}}{{end}}
+
+Usage:{{if and .Runnable (ne .UseLine "") (ne .UseLine $rootCmd)}}
+  {{if .HasFlags}}{{appendIfNotPresent .UseLine "[flags]"}}{{else}}{{.UseLine}}{{end}}{{end}}{{ if .HasSubCommands }}
+  {{ .CommandPath}} [command]
+
+Use "{{$rootCmd}} help <command>" for more information about a given command.{{end}}{{ if $optionsCmdFor}}
```

Comment:
`kubectl` uses the `--help` flag instead of the `help` command to suggest help. I'd suggest changing this to `Use "{{$rootCmd}} <command> --help" ...` instead. We did that recently in Origin BTW, you might want to get the latest versions of these files we have in master.  ;)